src/py/flwr/server/run_serverapp.py

```diff
@@ -162,10 +166,23 @@ def run_server_app() -> None:  # pylint: disable=too-many-branches
             driver_service_address=args.superlink,
             root_certificates=root_certificates,
         )
-        flwr_dir = get_flwr_dir(args.flwr_dir)
         run_ = driver.run
-        app_path = str(get_project_dir(run_.fab_id, run_.fab_version, flwr_dir))
-        config = get_project_config(app_path)
+        if run_.fab_hash:
+            fab_req = GetFabRequest(hash_str=run_.fab_hash)
+            # pylint: disable-next=W0212
+            fab_res: GetFabResponse = driver._stub.GetFab(fab_req)
+            if fab_res.fab.hash_str != run_.fab_hash:
+                raise ValueError("FAB hashes don't match.")
+
+            config = get_fab_config(fab_res.fab.content)
+            server_app_run_config = fuse_dicts(config, run_.override_config)
```

Comment:
This looks like it downloads the FAB only for the default config, right? Shouldn't it also install the FAB and then later load the ServerApp from it?